### PR TITLE
docs: update RangePicker month demo

### PR DESCRIPTION
**Pull request by zombieJ**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

ref #48835 #48590

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      -     |
| 🇨🇳 Chinese |      -     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Review by li-jia-nan — APPROVED** [review]

File: .dumi/pages/index/components/Banner.tsx
Comment (kiner-tang) [inline .dumi/pages/index/components/Banner.tsx:1]: 为啥要删掉呀？

Comment (li-jia-nan) [reply .dumi/pages/index/components/Banner.tsx:1]: 我丢，我都没看到删了个文件

File: components/date-picker/demo/select-in-range.tsx [hunk 1]
```diff
@@ -1,24 +1,37 @@
...
-    }
+  return false;
+};
 
-    return false;
-  };
+// Disabled 6 months from the selected date
+const disabled6MonthsDate: DatePickerProps['disabledDate'] = (current, { from }) => {
+  if (from) {
+    const curMonth = current.year() * 12 + current.month();
```
Comment (crazyair) [inline components/date-picker/demo/select-in-range.tsx:19]: 年为什么要乘以12

File: .dumi/pages/index/components/Banner.tsx
Comment (kiner-tang) [reply .dumi/pages/index/components/Banner.tsx:1]: 看起来是改用 PreviewBanner 了，以前的 Banner 确实没有地方在用了
image

**Review by kiner-tang — APPROVED** [review]

File: components/date-picker/demo/select-in-range.tsx (hunk 1 above)
Comment (zombieJ) [reply components/date-picker/demo/select-in-range.tsx:19]: 因为一年 12 个月……

File: .dumi/pages/index/components/Banner.tsx
Comment (zombieJ) [reply .dumi/pages/index/components/Banner.tsx:1]: 嗯，提示用不到 warning 了所以删了。

**Pull request merged**